components/Pages/Grants/ExternalId/ExternalIds.tsx

```diff
@@ -0,0 +1,169 @@
+"use client";
+import Link from "next/link";
+import { AddExternalId } from "./AddExternalIdDialog";
+import { useState, useEffect } from "react";
+import fetchData from "@/utilities/fetchData";
+import { INDEXER } from "@/utilities/indexer";
+import toast from "react-hot-toast";
+import { TrashIcon } from "@heroicons/react/24/outline";
+
+export default function ExternalIds({
+  projectUID,
+  communityUID,
+  externalIds,
+}: {
+  projectUID: string;
+  communityUID: string;
+  externalIds: string[];
+}) {
+  // Mock data
+
+  const [gitcoinUrls, setGitcoinUrls] = useState<{ [key: string]: string[] }>(
+    {}
+  );
+
+  const [removingId, setRemovingId] = useState<string | null>(null);
+
+  const handleRemove = async (id: string) => {
+    setRemovingId(id);
+    const [result, error] = await fetchData(
+      INDEXER.GRANTS.REMOVE_EXTERNAL_ID,
+      "POST",
+      {
+        projectUID: projectUID,
+        communityUID: communityUID,
+        externalId: id,
+      },
+      {},
+      {},
+      true
+    );
+    if (error) {
+      toast.error("Error removing external ID");
+    } else {
+      toast.success("External ID removed successfully");
+    }
+    setRemovingId(null);
+  };
+
+  async function fetchApplicationsByProjectId(projectId: string) {
+    try {
+      const url = "https://grants-stack-indexer-v2.gitcoin.co/graphql";
+
+      const payload = {
+        query: `
+      query MyQuery {
+        applications(condition: {projectId: "${projectId}"}) {
+          chainId
+          roundId
+          id
+        }
+      }
+    `,
+      };
+
+      const response = await fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(payload),
+      });
+
+      const data = await response.json();
+      const applications = data.data.applications;
+
+      if (applications && applications.length > 0) {
+        return applications.map(
+          ({
+            chainId,
+            roundId,
+            id: applicationId,
+          }: {
+            chainId: string;
+            roundId: string;
+            id: string;
+          }) =>
+            `https://explorer.gitcoin.co/#/round/${chainId}/${roundId}/${applicationId}`
+        );
+      }
+      return [];
+    } catch (error) {
+      console.error("Error fetching applications:", error);
+      return [];
+    }
+  }
+
+  useEffect(() => {
+    const fetchAllUrls = async () => {
+      const urls: { [key: string]: string[] } = {};
+      for (const externalId of externalIds) {
+        const applicationUrls = await fetchApplicationsByProjectId(externalId);
+        if (applicationUrls.length > 0) {
+          urls[externalId] = applicationUrls;
+        }
+      }
+      setGitcoinUrls(urls);
+    };
+
+    fetchAllUrls();
+  }, [externalIds]);
```

Comment:
i think you can use useQuery hook for it

Reply:
Done!